x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/use_rules.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+
+import { useKibanaUiSetting } from '../../../lib/settings/use_kibana_ui_setting';
+import { DEFAULT_KBN_VERSION } from '../../../../common/constants';
+import { FetchRulesResponse, FilterOptions, PaginationOptions } from './types';
+import { useStateToaster } from '../../../components/toasters';
+import { fetchRules } from './api';
+import { errorToToaster } from '../../../components/ml/api/error_to_toaster';
+import * as i18n from './translations';
+
+type Return = [boolean, FetchRulesResponse];
+
+/**
+ * Hook for using the list of Rules from the Detection Engine API
+ *
+ * @param pagination desired pagination options (e.g. page/perPage)
+ * @param filterOptions desired filters (e.g. filter/sortField/sortOrder)
+ * @param refetchToggle toggle for refetching data
+ */
+export const useRules = (
+  pagination: PaginationOptions,
+  filterOptions: FilterOptions,
+  refetchToggle: boolean
+): Return => {
+  const [rules, setRules] = useState<FetchRulesResponse>({
+    page: 1,
+    perPage: 20,
+    total: 0,
+    data: [],
+  });
+  const [loading, setLoading] = useState(true);
+  const [kbnVersion] = useKibanaUiSetting(DEFAULT_KBN_VERSION);
+  const [, dispatchToaster] = useStateToaster();
+
+  useEffect(() => {
+    let isSubscribed = true;
+    const abortCtrl = new AbortController();
+    setLoading(true);
+
+    async function fetchData() {
+      try {
+        const fetchRulesResult = await fetchRules({ filterOptions, pagination, kbnVersion });
```

Comment:
Shouldn't you be passing down the abortCtrl to fetchRules?

Reply:
Nice catch -- adding! 🙂